### DISCUSSION TITLE
Disaster recovery

### DISCUSSION THREAD

**Discussion by ErikLundJensen** [crossplane/crossplane#2244] [General]
Hi,
I'm looking into creating Kubernetes cluster using Crossplan and populate clusters using GitOps (Argo CD).
Let's assume I have a CrossPlane enabled cluster A which creates cluster B.

What will happen if I destroy cluster A and re-install cluster A again?

The same CrossPlane custom resources will be created in cluster A, but how can I'm associate the existing cluster B with the newly created CrossPlane resources?

best regards
Erik

**Comment by muvaf** [comment]
You can import an existing resource by using its identifier in `crossplane.io/external-name` annotation. If you create a managed resource with that annotation, this will be used for creation and then when you deploy same managed resource in another cluster, Crossplane will find that resource and reconcile it. If you don't give a value for that annotation and the provider allows users to name the resource, then the `metadata.name` of the managed resource will be copied to `crossplane.io/external-name` in the first reconciliation.

However, if the provider doesn't let users name the resource, like AWS VPC, the controller saves the identifier to `crossplane.io/external-name` right after its creation in the provider. So you need to commit that value in your git repo so that in another cluster, Crossplane sees that the value exists and try to reconcile it. Otherwise, if it's empty, it will not copy `metadata.name` since resource cannot be named and when `crossplane.io/external-name` is empty Crossplane will try to create a new resource.
